src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseFileServiceTests.cs

```diff
@@ -2606,6 +2608,120 @@ await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId)
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var newFormFile = CreateFormFileMock("newAncillary.pdf", "application/pdf").Object;
+            var blobStorageService = new Mock<IBlobStorageService>(Strict);
+            var fileUploadsValidatorService = new Mock<IFileUploadsValidatorService>(Strict);
+
+            blobStorageService.Setup(mock =>
+                mock.UploadFile(PrivateReleaseFiles,
+                    It.Is<string>(path =>
+                        path.Contains(FilesPath(release.Id, Ancillary))),
+                    newFormFile
+                )).Returns(Task.CompletedTask);
+
+            fileUploadsValidatorService.Setup(mock =>
+                    mock.ValidateFileForUpload(newFormFile, Ancillary))
+                .ReturnsAsync(Unit.Instance);
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var service = SetupReleaseFileService(contentDbContext: contentDbContext,
+                    blobStorageService: blobStorageService.Object,
+                    fileUploadsValidatorService: fileUploadsValidatorService.Object);
+
+                var result = await service.ReplaceAncillary(
+                    release.Id,
+                    releaseFile.FileId,
+                    newFile: newFormFile
+                );
+
+                MockUtils.VerifyAllMocks(blobStorageService, fileUploadsValidatorService);
+
+                var fileInfo = result.AssertRight();
+
+                fileUploadsValidatorService.Verify(mock =>
+                    mock.ValidateFileForUpload(newFormFile, Ancillary), Times.Once);
+
+                blobStorageService.Verify(mock =>
+                    mock.UploadFile(PrivateReleaseFiles,
+                        It.Is<string>(path =>
+                            path.Contains(FilesPath(release.Id, Ancillary))),
+                        newFormFile
+                    ), Times.Once);
+
+                Assert.True(fileInfo.Id.HasValue);
+                Assert.Equal("pdf", fileInfo.Extension);
+                Assert.Equal("newAncillary.pdf", fileInfo.FileName);
+                Assert.Equal("Ancillary name", fileInfo.Name);
+                Assert.Equal("Ancillary summary", fileInfo.Summary);
+                Assert.Equal("10 Kb", fileInfo.Size);
+                Assert.Equal(Ancillary, fileInfo.Type);
+                Assert.Equal("test@test.com", fileInfo.UserName);
+                Assert.InRange(DateTime.UtcNow.Subtract(fileInfo.Created.GetValueOrDefault()).Milliseconds, 0, 1500);
+            }
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                // Old File shouldn't be removed
+                Assert.Single(await contentDbContext.Files
+                    .Where(f => f.Id == releaseFile.FileId)
+                    .ToListAsync());
```

Comment:
Whilst this is probably incorrect behaviour (see other comment on deleting the old file), it'd probably be cleaner to check that something is missing by using an assertion like:

```cs
Assert.NotNull(await contentDbContext.Files
    .FirstOrDefaultAsync(f => f.Id == releaseFile.FileId));
```